sponsorship/src/test/java/com/kyriba/conference/sponsorship/service/KafkaEmailClientImplTest.java

```diff
@@ -0,0 +1,79 @@
+package com.kyriba.conference.sponsorship.service;
+
+import com.kyriba.conference.sponsorship.configuration.KafkaConfiguration;
+import com.kyriba.conference.sponsorship.domain.EmailMessage;
+import com.kyriba.conference.sponsorship.domain.Sponsor;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@ExtendWith({ SpringExtension.class })
+@SpringBootTest(classes = { KafkaEmailClientImpl.class, KafkaConfiguration.class })
+@EmbeddedKafka(partitions = 1, topics = { "NotificationTopic" },
```

Comment:
I recommend to use TestContainers for Kafka and other middleware in tests, this is more realistic approach

Reply:
I'll migrate to testcontainers in a separate pull request. It appeared to be NOT easy(( It is not possible to use custom port for kafka testcontainer, so I need somehow start testcontainer, get randomly generated port, create my producer with this port, but my producer use the port from `application.yml`... I don't know how to resolve it